### PR TITLE
add SyncBatchNorm to docs

### LINKED ISSUE #18983: [docs] nn.SyncBatchNorm missing in online docs
`nn.SyncBatchNorm` docs are missing from master [online docs](https://pytorch.org/docs/master/search.html?q=SyncBatchNorm&check_keywords=yes&area=default#), despite [doc string](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/batchnorm.py#L322-L406) in code.

Indeed, no SyncBatchNorm in [nn.rst](https://github.com/pytorch/pytorch/blob/master/docs/source/nn.rst#batchnorm1d).

A proposal: making a public docs html page "DocTODO" that lists all public classes (without underscore in name) that are missing from rst files and maybe functions as well, along with some color mark if doc string is already present. Like this it's super easy to find these cases and renders pointless creating issues like this one.

### PR DESCRIPTION
**Pull request by soumith**
Fixes https://github.com/pytorch/pytorch/issues/18983

### CODE REVIEW

**Comment by vadimkantorov** [comment]
also there are a lot of open issues about sync batch norm, not sure if they need to be closed: e.g. https://github.com/pytorch/pytorch/issues/2584
